pkg/azclient/auth.go

```diff
@@ -133,20 +169,58 @@ func NewAuthProvider(armConfig *ARMClientConfig, config *AzureAuthConfig, client
 		if err != nil {
 			return nil, fmt.Errorf("decoding the client certificate: %w", err)
 		}
-		computeCredential, err = azidentity.NewClientCertificateCredential(armConfig.GetTenantID(), config.GetAADClientID(), certificate, privateKey, credOptions)
-		if err != nil {
-			return nil, err
-		}
+
 		if IsMultiTenant(armConfig) {
-			networkTokenCredential, err = azidentity.NewClientCertificateCredential(armConfig.NetworkResourceTenantID, config.GetAADClientID(), certificate, privateKey, credOptions)
-			if err != nil {
-				return nil, err
+
+			// Network credential for network resource access
+			{
+				credOptions := &azidentity.ClientCertificateCredentialOptions{
+					ClientOptions:        *clientOption,
+					SendCertificateChain: true,
+				}
+				networkCredential, err = azidentity.NewClientCertificateCredential(
+					armConfig.NetworkResourceTenantID,
+					config.GetAADClientID(),
+					certificate,
+					privateKey,
+					credOptions,
+				)
+				if err != nil {
+					return nil, err
+				}
 			}
-			credOptions = &azidentity.ClientCertificateCredentialOptions{
-				ClientOptions:              *clientOption,
-				AdditionallyAllowedTenants: []string{armConfig.NetworkResourceTenantID},
+
+			// Compute credential with additional allowed tenants for cross-tenant access
+			{
+				credOptions := &azidentity.ClientCertificateCredentialOptions{
+					ClientOptions:              *clientOption,
+					AdditionallyAllowedTenants: []string{armConfig.NetworkResourceTenantID},
+					SendCertificateChain:       true,
```

Comment:
@MartinForReal Kindly note that I have also add the missing option here. please take a look, thanks.